actionlist/app/reducers/action_items.js

```diff
@@ -0,0 +1,28 @@
+/* eslint import/no-extraneous-dependencies: ["error", {"peerDependencies": true}] */
+import { handleActions } from 'redux-actions';
+import * as types from '../actions/types';
+
+const lib = require('../lib/commons.js');
+
+const list = lib.getHashJSON();
+
+export const defaultState = {
+  dataSource: list,
+};
+
+const toggleItem = (state, action) => {
+  const oldItem = state.dataSource[action.payload];
+  const newItem = oldItem;
+  if (newItem) {
+    newItem.expanded = !newItem.expanded;
+  }
+  return {
+    ...state,
+    dataSource: state.dataSource,
+    oldItem: newItem,
```

Comment:
I don't think this function is doing what you think it's doing. What it probably ought to be doing is 
```javascript
const toggleItem = (state, action) => {
  const dataSource = state.dataSource;
  const oldItem = dataSource[action.payload];
  const newItem = {
    ...oldItem,
    expanded: !oldItem.expanded
  };

  return {
    ...state,
    dataSource: {
      ...dataSource,
      [action.payload]: newItem
    }
  };
};
```
?
This is where having unit tests on your reducers would be handy.